### DISCUSSION TITLE
Dropdown in Stencil to filter the elements based on type

### DISCUSSION THREAD

**Discussion by kvinita** [clientIO/joint#2533] [Q&A] [answered]
### Introduction

I have a stencil containing over 100+ elements of different types. I would like to implement a dropdown feature within the stencil to enable selection of specific types of elements to be displayed. Additionally, the dropdown should support multi-select functionality. Is this feature currently supported in JointJS+?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
You can add an independent select box above the stencil and programmatically control the content of the stencil when its value changes.

To load elements into the stencil, use the `stencil.load()` method. Depending on the use case, but you can also load all elements first and then filter them using the `stencil.filter()` method.

Here's an example how to add a select box to every stencil group: https://codepen.io/jointjs/full/mdaNOPg
